### ISSUE TITLE
코드 수정 사항 (작성중)

### ISSUE THREAD

**Issue by jang1suh** [SNU-DRL/Attribution-ECG#14]
공통)
- Unused imports 제거해 주세요. Sort imports 해주세요.
- Function에 unused arguments 제거해 주세요.
- argument parser help 문장, 주석이 이해가 쉽게 적혀있는지 확인해 주세요.
- 나중에 공개하면 README도 작성해야 할텐데, 시간 있을 때 미리 작성해두면 나중에 편할 듯합니다.
- 아래 적은 내용 외에도 최종적으로 결정된 사항 반영하여 코드 확인 및 수정 부탁드립니다.
- 마지막으로 실험 돌리기 전에 버그 없는지 확인해주세요!

---

- ReLU, Flatten을 forward 안에서 정의하고 사용하는 것으로 바꾸느라 new_resnet1d.py 파일을 새로 만들었네요. 기존 resnet1d.py 파일을 삭제하고 현재 new_resnet1d.py 파일을 resnet1d.py로 변경해 주세요.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/models/__init__.py#L1-L2

- scheduler가 정의되고 사용되고 있지는 않습니다. scheduler를 사용하지 않으려면 없애고 scheduler를 사용할 것이라면 scheduler.step을 optimizing loop에 추가해 주어야 할 듯합니다. (scheduler를 꼭 사용할 필요는 없을 것 같기는 합니다)
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/train.py#L80

- test set의 결과를 보면서 best model을 고르기 때문에 overfitting입니다. epoch을 정해놓고 마지막 모델을 사용하는 편이 좋을 듯합니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/train.py#L87

- localization error -> localization score로 명칭이 바뀌었습니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/eval_attribution.py#L29

- perburbation-based -> degradation score로 명칭이 바뀌었습니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/eval_attribution.py#L53

- loc_score, pnt_score, deg_score로 바꿔주어야 이해가 쉬울 듯합니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/eval_attribution.py#L120-L126

- 여기도 loc, pnt, deg로 바꿔주세요.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L144-L151

- accuracy는 사용하지 않으므로 iou만 남기는 편이 좋을 듯합니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/eval_attribution.py#L39-L40

- 여기도 IOU만 남기면 됩니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L166-L175

- boundaries per label은 적절하지 않은 구인듯합니다. boundaries by label이 더 적절할 듯합니다. 이 소스코드에 있는 boundaries per label을 모두 boundaries by label로 바꿔야 이해가 쉬울 듯합니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/src/eval_attribution.py#L26

- disease_idx가 아니라 class_idx로 바꿔주세요.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L254

- feature_ablation은 사용하지 않기로 했으니 목록에서 지워주세요.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L83

- confidence가 높은 sample만 보기로 했으니 이 옵션도 필요 없어졌습니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L27

- 여기도 correct 옵션이 필요 없을듯 합니다.
https://github.com/SNU-DRL/Attribution-ECG/blob/1be046c2c4ce60f5cbac091c2d5a832863b281bf/evaluate_attributions.py#L231
